syntax = "proto3";

package assfire.api.v1.scheduler;
option go_package = "assfire.org/api/v1/scheduler";

import "assfire/api/v1/scheduler/concepts.proto";
import "google/api/annotations.proto";

message ScheduleWaybillRequest {
  Waybill waybill = 1;
  WaybillSchedulingAlgorithmType algorithm_type = 2;
}

message ScheduleWaybillResponse {
  message Status {
    enum Code {
      STATUS_CODE_UNSPECIFIED = 0;
      STATUS_CODE_OK = 1;
      STATUS_CODE_ERROR = 2;
    }
    Code code = 1;
    string message = 2;
  }
  Waybill waybill = 1;
  Status status = 2;
}

service SchedulerService {
  rpc ScheduleWaybill(ScheduleWaybillRequest) returns (ScheduleWaybillResponse) {
    option (google.api.http) = {
      get: "/v1/schedule/waybill"
    };
  }
}